src/bloqade/pyqrack/squin/qubit.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any
+
+from kirin import interp
+from kirin.dialects import ilist
+from kirin.interp.exceptions import InterpreterError
+
+from bloqade.squin import qubit
+from bloqade.pyqrack.reg import QubitState, PyQrackQubit
+from bloqade.pyqrack.base import PyQrackInterpreter
+
+from .runtime import OperatorRuntimeABC
+
+
+@qubit.dialect.register(key="pyqrack")
+class PyQrackMethods(interp.MethodTable):
+    @interp.impl(qubit.New)
+    def new(self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: qubit.New):
+        n_qubits: int = frame.get(stmt.n_qubits)
+        qreg = ilist.IList(
+            [
+                PyQrackQubit(i, interp.memory.sim_reg, QubitState.Active)
+                for i in interp.memory.allocate(n_qubits=n_qubits)
+            ]
+        )
+        return (qreg,)
+
+    @interp.impl(qubit.Apply)
+    def apply(self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: qubit.Apply):
+        qubits: ilist.IList[PyQrackQubit, Any] = frame.get(stmt.qubits)
+
+        for qbit in qubits:
+            if not qbit.is_active():
+                return ()
+
+        operator: OperatorRuntimeABC = frame.get(stmt.operator)
+        operator.apply(*qubits)
+
+    @interp.impl(qubit.Broadcast)
+    def broadcast(
+        self, interp: PyQrackInterpreter, frame: interp.Frame, stmt: qubit.Broadcast
+    ):
+        operator: OperatorRuntimeABC = frame.get(stmt.operator)
+        qubits: ilist.IList[PyQrackQubit, Any] = frame.get(stmt.qubits)
+        operator.broadcast_apply(qubits)
+
+    def _measure_qubit(self, qbit: PyQrackQubit):
+        if qbit.is_active():
+            return qbit.sim_reg.m(qbit.addr)
```

Comment:
```suggestion
            return bool(qbit.sim_reg.m(qbit.addr))
```
this returns 0, 1 as an integer.